src/lang/base/BuiltIns.ml

```diff
@@ -984,12 +985,13 @@ module ScillaBuiltIns
             (* Second: check applicability *)
             let%bind res_type = fun_type_applies type_elab argtypes in
             pure (type_elab, res_type, exec)
-          else fail @@ "Name or arity don't match") in
+          else fail0 @@ "Name or arity don't match") in
       let open Caml in
       let dict = Option.value ~default:[] @@ Hashtbl.find_opt built_in_hashtbl opname in
       let%bind (_, (type_elab, res_type, exec)) = tryM dict ~f:finder
-          ~msg:(sprintf "[%s] Cannot find built-in with name \"%s\" and argument types %s."
-                  (ER.get_loc (get_rep op) |> get_loc_str) opname (pp_typ_list argtypes))
+          ~msg:(mk_error1
+                (sprintf "Cannot find built-in with name \"%s\" and argument types %s." opname (pp_typ_list argtypes))
```

Comment:
We really should turn the `msg` parameter into a thunk that is only evaluated if we need to print the error message. The pattern we are using is bound to cause performance problems again somewhere down the line.

I'll create a separate issue on it, so don't worry about it for this PR.

Reply:
Thank you Jacob for creating #270 .